src/browser.ts

```diff
@@ -98,11 +98,16 @@ function flushBuffered(analytics: Analytics): void {
       // @ts-expect-error
       typeof analytics[operation] === 'function'
     ) {
-      // flush each individual event as its own task, so not to block initial page loads
-      setTimeout(() => {
+      if (operation === 'addSourceMiddleware') {
         // @ts-expect-error
-        analytics[operation].call(analytics, ...args)
-      }, 0)
+        await analytics[operation].call(analytics, ...args)
```

Comment:
Just for simplicity, I wonder if this still works fine without using async/await. For example, just run `addSourceMiddleware` without the `setTimeout`, and then let the rest of operations run async in their own task.

Reply:
That was my first thought as well but calling addSourceMiddleware without the setTimeout was still resulting in the page call being fired first 🤔

Reply:
Oh, I see what's going on. Since we have a separate chunk for the `middleware` plugin, then the `analytics.addSourceMiddleware` happens asynchronously, because it has to lazy-load the middleware first.